geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/QueryFilterSplitter.scala

```diff
@@ -130,10 +146,99 @@ class QueryFilterSplitter(sft: SimpleFeatureType) extends LazyLogging {
     if (options.isEmpty) {
       options.append(fullTableScanOption(andOption(validFilters)))
     }
-    options.toSeq
+
+    options
   }
 
-  private def getOrQueryOptions(filter: Or): Seq[FilterPlan] = {
+  /**
+    * Get options for a filter which contain an 'OR'
+    *
+    * @param filter original filter as passed in
+    * @param dnf same filter in disjunctive normal form - OR will be only at the top of the filter tree
+    * @param cnf same filter in conjunctive normal form - AND will be only at the top of the filter tree
+    * @return
+    */
+  private def getOrQueryOptions(filter: Filter, dnf: Or, cnf: Filter): Seq[FilterPlan] = {
+
+    import QueryFilterSplitter.isFullTableScan
+
+    // check for basic ORs with no nested ANDs/Ors
+    // e.g a  = '1' OR bbox OR during
+    def basicOrs: Option[Seq[FilterPlan]] = Some(cnf).collect { case o: Or => o.getChildren }.map { ors =>
+      // detect single attribute queries like "name in ('1', '2')"
+      val attributes = ors.flatMap(getAttributeProperty).map(_.name)
+      if (attributes.length == ors.length && attributes.forall(_ == attributes.head)) {
+        Seq(FilterPlan(Seq(getSameAttributeOption(ors))))
+      } else {
+        // get the option for each piece of the OR and combine them
+        val orOptions = ors.map(f => getAndQueryOptions(Seq(f)))
+        if (orOptions.exists(opt => opt.length == 1 && opt.head.filters.exists(isFullTableScan))) {
+          // if we have to do a full table scan, we can just compress everything into that
+          Seq(fullTableScanOption(Some(filter)))
+        } else {
+          // multiple attributes being queried, come up with the different permutations of options
+          // note: this is O(2^n) but at most will usually only have 1-2 options
+          orOptions.reduce { (left, right) => left.flatMap(l => right.map(r => FilterPlan(l.filters ++ r.filters))) }
+        }
+      }
+    }
+
+    // check for attribute ORs that are ANDed with non-attribute filters
+    // e.g. a in ('1', '2') AND bbox AND during
+    def attributeOrs: Option[Seq[FilterPlan]] = Some(cnf).collect { case a: And => a }.flatMap { a =>
+      val (orFilters, nonOrs) = a.getChildren.partition(_.isInstanceOf[Or])
+      // we only deal with a single OR here
+      val singleOr = Some(orFilters).collect {case Seq(o: Or) => o.getChildren }
+      // make sure that the ORs are on a single attribute, and not a spatio-temporal one
+      val singleAttributeOr = singleOr.filter { ors =>
+        // TODO this won't take into accound indexed dates... but our getSameAttributeOption doesn't either
```

Comment:
fixed spelling and opened a ticket https://geomesa.atlassian.net/browse/GEOMESA-1168